### ISSUE TITLE
Billing email address reverting to old email address on checkout

### ISSUE THREAD

**Issue by mattgetmilk** [woocommerce/woocommerce#17011]
- [ x] I have searched for similar issues in both open and closed tickets and cannot find a duplicate
- [ ] The issue still exists against the latest `master` branch of WooCommerce on Github (this is **not** the same version as on WordPress.org!)
- [x ] I have attempted to find the simplest possible steps to reproduce the issue
- [ ] I have included a failing test as a pull request (Optional)

## Steps to reproduce the issue

I have prepared a default install at http://woocommerce.getmilk.com.au/wordpress

Admin: getmilk
Password: getmilk

User for testing (I have already done the email update with this user)

Username: info@luvkins.com.au
P/W: getmilk

1. Make a purchase and register to site as a new user
2. Once transaction is complete, edit email address under my-account and addresses
3. make another purchase - note that the billing address email has reverted back to your old one

**LIKEWISE with your first and last name. Backend shows that the First and Last name have been updated, however internally they are still the old first and last name.**

## Expected/actual behavior

When I follow those steps, I see that the new email address I changed to is not the email address pre populated in the billing email address. Problem this causes, is that the user will process the order with the 'old' email address, which then sets the email back to the old one. FYI, the email address in backend is actually updated??

I was expecting to see...that the email address updates throughout wp/wc. I have tried this on numerous installs, themes, etc.

## Isolating the problem

- [ ] This bug happens with only WooCommerce plugin active
- [ x] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/)
- [x ] I can reproduce this bug consistently using the steps above

## WordPress Environment

<details>
```
`
### WordPress Environment ###

Home URL: http://woocommerce.getmilk.com.au/wordpress
Site URL: http://woocommerce.getmilk.com.au/wordpress
WC Version: 3.1.2
Log Directory Writable: ✔
WP Version: 4.8.2
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: –
WP Cron: ✔
Language: en_US

### Server Environment ###

Server Info: LiteSpeed
PHP Version: 5.6.31
PHP Post Max Size: 64 MB
PHP Time Limit: 300
PHP Max Input Vars: 1000
cURL Version: 7.54.0
OpenSSL/1.0.2k

SUHOSIN Installed: –
MySQL Version: 5.6.37
Max Upload Size: 32 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 3.1.2
WC Database Prefix: wp_
woocommerce_sessions: ✔
woocommerce_api_keys: ✔
woocommerce_attribute_taxonomies: ✔
woocommerce_downloadable_product_permissions: ✔
woocommerce_order_items: ✔
woocommerce_order_itemmeta: ✔
woocommerce_tax_rates: ✔
woocommerce_tax_rate_locations: ✔
woocommerce_shipping_zones: ✔
woocommerce_shipping_zone_locations: ✔
woocommerce_shipping_zone_methods: ✔
woocommerce_payment_tokens: ✔
woocommerce_payment_tokenmeta: ✔
MaxMind GeoIP Database: ❌ The MaxMind GeoIP Database does not exist - Geolocation will not function. You can download and install it manually from http://dev.maxmind.com/geoip/legacy/geolite/ to the path: . Scroll down to "Downloads" and download the "Binary / gzip" file next to "GeoLite Country". Please remember to uncompress GeoIP.dat.gz and upload the GeoIP.dat file only.

### Security ###

Secure connection (HTTPS): ❌Your store is not using HTTPS. Learn more about HTTPS and SSL Certificates.
Hide errors from visitors: ✔

### Active Plugins (1) ###

WooCommerce: by Automattic – 3.1.2

### Settings ###

API Enabled: ✔
Force SSL: –
Currency: AUD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)


### WC Pages ###

Shop base: #4 - /index.php/shop/
Cart: #5 - /index.php/cart/
Checkout: #6 - /index.php/checkout/
My account: #7 - /index.php/my-account/

### Theme ###

Name: Storefront
Version: 2.2.5
Author URL: https://woocommerce.com/
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ✔

### Templates ###

Overrides: –
`
```
</details>

**Comment by mikejolley** [comment]
This data is stored separately.

- `/my-account/edit-account/` is the WP user account name + account email.
- `/my-account/edit-address/billing/` is for the billing address, which also includes an email address.

Checkout billing fields will show the billing address.

3.2 (coming in the next few weeks) has some updated handling to update the user session when the addresses are updated in my-account. This should fix the issue for you.

Note: please reset that admin password, this is a public issue tracker.

**Comment by mattgetmilk** [comment]
Hi Mike, thanks for the reply. So when the user updates their billing information as well, this will also ensure that the correct details are returned to the checkout page? I have tried clearing cache etc which unfortunately still displays an old email address and name. From what we can see, there are two tables that seem to keep updated information and previous account/billing data. However, the 'wrong' table may be being used to grab this data on checkout. Maybe that helps. Again, thanks for looking into this. Re password, happy to leave if you guys need access to my install for the time being.

**Comment by dougaitken** [comment]
Hey @mattgetmilk, about to reply to your ticket. Best to remove those login details completely. We can get access via the ticket conversation if needed 👍

**Comment by mikejolley** [comment]
@mattgetmilk The issue is caused by the current session, which 3.2 fixes. So if you update an address, the session data is updated also.
